File: problems/Coin Change 2/change.java
class Solution {
    public int change(int amount, int[] coins) {
        int[] dp = new int[amount + 1];

        dp[0] = 1;

        for (int coin : coins) {
            for (int item = coin; amount >= item; ++item) {
                dp[item] += dp[item - coin];
            }
        }

        return dp[amount];
    }
}

File: problems/Coin Change 2/change.cs
public class Solution {
    public int Change(int amount, int[] coins) {
        int[] dp = new int[1 + amount];

        dp[0] = 1;

        foreach (int coin in coins) {
            for (int sum = 1; amount >= sum; ++sum) {
                if (sum >= coin) {
                    dp[sum] += dp[sum - coin];
                }
            }
        }

        return dp[amount];
    }
}
